docs/General-Workflow.md

```diff
@@ -115,43 +122,43 @@ The linker script links
 - `.data` (only) from the .o built from `energy_orb_wave.data.s`
 - `.rodata` (only) from the .o built from `energy_orb_wave.rodata.s`
 
-.data (respectively .rodata) is not linked from the .o built from `energy_orb_wave.c`, because the subsegments include segments with the `data` (respectively `rodata`) segment type
-
 # 4 Decompile data
 
-Move (decompile) data/rodata to the .c, using structs or relying on strings used in the code, or other things
+TODO: `mention migrate_rodata_to_functions: yes` functionality.
 
-Again, the .data/.rodata sections from the .o built from the .c will not be linked as long as there are any `data`/`rodata` subsegment in the code segment (and not just for `energy_orb_wave`, any other subsegment too)
+Move (decompile) data to the .c file, using structs or relying on strings used in the code.
 
-To link the .data/.rodata from the .o built from the .c (instead of from the .s files), the subsegments should be changed from
+To link the .data/.rodata from the .o built from the .c (instead of from the .s files), the subsegments should be changed from:
 
 ```yaml
 - [0x42100, c, energy_orb_wave]
-- [0x42200, data, energy_orb_wave]
-- [0x42300, rodata, energy_orb_wave]
+- [0x42200, data, energy_orb_wave]     #  extract data at this ROM address as energy_orb_wave.data.s
+- [0x42300, rodata, energy_orb_wave]   # extract rodata at this ROM address as energy_orb_wave.rodata.s
 ```
 
-to
+to:
 
 ```yaml
 - [0x42100, c, energy_orb_wave]
-- [0x42200, .data, energy_orb_wave]
-- [0x42300, .rodata, energy_orb_wave]
+- [0x42200, .data, energy_orb_wave]    # take the .data section from the compiled c file named energy_orb_wave
+- [0x42300, .rodata, energy_orb_wave]  # take the .rodata section from the compiled c file named energy_orb_wave
 ```
 
-If using `auto_all_section` and there is no other `data`/`.data`/`rodata`/`.rodata` in the subsegments in the code segment, the subsegments can also be changed to
+If using `auto_all_section` and there are no other `data`/`.data`/`rodata`/`.rodata` in the subsegments in the code segment, the subsegments can also be changed to
 
 ```yaml
 - [0x42100, c, energy_orb_wave]
 - [0x42200]
 ```
 
-# 5 Done!
+# 5 Decompile bss
```

Comment:
This should mention bss sections require a special notation on the yaml, one that requires to specify the `vram` address of the section. Also the parent segment needs to specify the bss size of the whole segment